src/java.base/share/classes/java/util/Collections.java

```diff
@@ -1714,6 +1714,7 @@ public void remove() {
                         throw new UnsupportedOperationException();
                     }
                     public void forEachRemaining(Consumer<? super Map.Entry<K, V>> action) {
+                        Objects.requireNonNull(action);
                         i.forEachRemaining(entryConsumer(action));
```

Comment:
As pointed out in the bug report description, it might be better to add the `null` check to `entryConsumer`. That would avoid code duplication for the `null` checks for all the current callers of `entryConsumer`.

(This comment only applies here; for `CheckedEntrySet` below this does not work because it does not call `entryConsumer`.)

Reply:
Hello @Marcono1234, I prefer doing it earliest in the top level method (which specifies the contract) instead of doing it in an internal method that this method calls.